src/routes/(app)/strafbakken/+page.server.ts

```diff
@@ -0,0 +1,51 @@
+import type { Actions, PageServerLoad } from "./$types";
+import db from "$lib/server/db";
+import { getUser } from "$lib/server/userCache";
+import { fail } from "@sveltejs/kit";
+
+// Load een overview van alle strafbakken
+export const load = (async () => {
+  return {
+    strafbakken: await db.user.findMany({
+      select: {
+        firstName: true,
+        nickname: true,
+        id: true,
+        _count: {
+          select: {
+            StrafbakReceived: {
+              where: {
+                dateDeleted: null,
+              },
+            },
+          },
+        },
+      },
+    }),
+  };
+}) satisfies PageServerLoad;
+
+export const actions = {
+  // Geef een strafbak
+  default: async ({ request, locals }: { request: Request; locals: any }) => {
+    const data = await request.formData();
+
+    const reason = data.get("reason")?.toString() || undefined;
+    const receiverId = Number(data.get("receiver"));
+
+    // Probeer de user te vinden als dat niet lukt om de een of andere reden, dan is het onsuccesvol
+    const session = await locals.getSession();
+    const user = await getUser(session);
+    const giverId = user?.id;
+    if (!giverId) return fail(400);
+
+    await db.strafbak.create({
+      data: {
+        giverId,
+        receiverId,
+        reason,
+      },
+    });
```

Comment:
kan dit ook stuk gaan? als in wordt `.catch()` ooit gecalled?

Reply:
ja dit kan je wel vrij makkelijk slopen, receiverId krijgt hij uit een hidden input en kan je dus zo veranderen naar een niet bestaande id wat een error geeft. Een andere wel bestaand id maakt overigens niet uit want iedereen kan toch al iedereen een bak geven.  Hier moet dus nog wel even een try catch omheen